vault/rollback_test.go

```diff
@@ -241,35 +237,33 @@ func TestRollbackManager_WorkerPool(t *testing.T) {
 	done := make(chan struct{})
 
 	// start a goroutine to consume the remaining items from the queued work
+	gotAllPaths := make(chan struct{})
 	go func() {
+		channelClosed := false
+		gotLock.Lock()
+		defer gotLock.Unlock()
 		for {
 			select {
 			case i := <-ran:
-				gotLock.Lock()
 				got[i] = true
-				gotLock.Unlock()
+				if len(got) == 10 && !channelClosed {
+					close(gotAllPaths)
```

Comment:
Why keep iterating here?

Reply:
there might be more queued rollback operations that need to complete as part of `core.rollback.Stop()`. This keeps them from blocking on a write to the `ran` channel, which is unbuffered.

Reply:
Makes sense, mind adding a little comment to that effect?